src/bun.zig

```diff
@@ -2816,6 +2816,22 @@ pub fn linuxKernelVersion() Semver.Version {
     return @import("./analytics.zig").GenerateHeader.GeneratePlatform.kernelVersion();
 }
 
+pub fn selfExePath() ![:0]u8 {
+    const memo = struct {
+        var set = false;
+        // TODO open zig issue to make 'std.fs.selfExePath' return [:0]u8 directly
+        // note: this doesn't use MAX_PATH_BYTES because on windows that's 32767*3+1 yet normal paths are 255.
+        // should this fail it will still do so gracefully. 4096 is MAX_PATH_BYTES on posix.
+        var value: [4096]u8 = undefined;
+        var len: usize = 0;
+    };
+    if (memo.set) return memo.value[0..memo.len :0];
+    const init = try std.fs.selfExePath(&memo.value);
+    memo.len = init.len;
+    memo.value[memo.len] = 0;
+    memo.set = true;
+    return memo.value[0..memo.len :0];
+}
```

Comment:
Are we sure that `selfExePath` will only be called from one thread?

Reply:
Fixed in c010351c3d46caba53e247a03a7c1f1840c3cee1